src/ut/alarm_trap_sender_test.cpp

```diff
@@ -0,0 +1,354 @@
+
+#include "fakenetsnmp.h"
+#include "fakelogger.h"
+#include "test_interposer.hpp"
+
+using ::testing::_;
+using ::testing::A;
+using ::testing::Return;
+using ::testing::ReturnNull;
+using ::testing::InSequence;
+using ::testing::MakeMatcher;
+using ::testing::Matcher;
+using ::testing::MatcherInterface;
+using ::testing::MatchResultListener;
+using ::testing::SaveArg;
+using ::testing::Invoke;
+
+class EnterpriseTrapVarsMatcher : public MatcherInterface<netsnmp_variable_list*>
+{
+public:
+  explicit EnterpriseTrapVarsMatcher(oid trap_type,
+                                     std::string MIB_version,
+                                     std::string name,
+                                     oid alarm_oid,
+                                     oid resource_id,
+                                     std::string severity,
+                                     std::string description,
+                                     std::string details,
+                                     std::string cause,
+                                     std::string effect,
+                                     std::string action) :
+      _trap_type(trap_type),
+      _MIB_version(MIB_version),
+      _name(name),
+      _alarm_oid(alarm_oid),
+      _resource_id(resource_id),
+      _severity(severity),
+      _description(description),
+      _details(details),
+      _cause(cause),
+      _effect(effect),
+      _action(action) {}
+
+  virtual bool MatchAndExplain(netsnmp_variable_list* vl,
+                               MatchResultListener* listener) const {
+    
+    oid trap_type = *(vl->val.objid);
+    vl = vl->next_variable;
+    std::string MIB_version(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable;
+    std::string name(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable;
+    oid alarm_oid = *(vl->val.objid);
+    vl = vl->next_variable;
+    oid resource_id = *(vl->val.objid);
+    vl = vl->next_variable;
+    std::string severity(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable;
+    std::string description(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string details(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string cause(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string effect(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string action(reinterpret_cast<char const*>(vl->val.string));
+
+    return (_description == description) && (_details == details) && (_cause == cause)
+           && (_effect == effect) && (_action == action) && (_resource_id == resource_id)
+           && (_alarm_oid == alarm_oid) && (_name == name) && (_MIB_version == MIB_version)
+           && (_trap_type == trap_type);
+  }
+
+  virtual void DescribeTo(::std::ostream* os) const {
+    *os << "trap is being sent";
```

Comment:
I've added the printable attributes of the alarm to the DescribeTo and DescribeNegationTo functions using a similar style to the RFC matcher. I've seen that DescribeTo gets printed when you fail a test but I'm not sure how DescribeNegationTo gets involved.